tests/Doctrine/Migrations/Tests/Configuration/Migration/ArrayLoaderTest.php

```diff
@@ -17,4 +20,42 @@ public function testInvalidKey() : void
         $loader = new ConfigurationArray(['foo' => 'aaa']);
         $loader->getConfiguration();
     }
+
+    public function testSchemaProviderInstance() : void
+    {
+        $schemaProvider = $this->createMock(SchemaProvider::class);
+        $loader         = new ConfigurationArray(['schema_provider' => $schemaProvider]);
+        $config         = $loader->getConfiguration();
+
+        self::assertSame($schemaProvider, $config->getSchemaProvider());
+    }
+
+    public function testSchemaProviderCallback() : void
+    {
+        $schemaProvider = $this->createMock(SchemaProvider::class);
+        $loader         = new ConfigurationArray([
+            'schema_provider' => static function () use ($schemaProvider) {
+                return $schemaProvider;
+            },
+        ]);
+        $config         = $loader->getConfiguration();
+
+        self::assertSame($schemaProvider, $config->getSchemaProvider());
+    }
+
+    public function testSchemaProviderString() : void
+    {
+        $schemaProvider = $this->createMock(SchemaProvider::class);
+        $loader         = new ConfigurationArray(['schema_provider' => get_class($schemaProvider)]);
+        $config         = $loader->getConfiguration();
+
+        self::assertInstanceOf(SchemaProvider::class, $config->getSchemaProvider());
+    }
+
+    public function testUnknownSchemaProvider() : void
+    {
+        $this->expectException(UnknownConfigurationValue::class);
+        $loader = new ConfigurationArray(['schema_provider' => 42]);
```

Comment:
For a complete coverage every check in the if block that leads to the exception would need to be tested.

Reply:
I am not sure what you are asking from me. The tests above cover all the allowed types. This test covers a non-supported type. Can you elaborate?